cumulusci/tasks/package_2gp.py

```diff
@@ -454,6 +461,45 @@ def _create_version_request(
         )
         return response["id"]
 
+    def _resolve_ancestor_id(self, spv_id: str = None) -> str:
+        """
+        If an ancestor_id (04t) is not specified, get it
+        from the latest production release.
+
+        @param spv_id The SubscriberPackageVersionId (04t) that is the ancestor
+        to the version being created.
+        """
```

Comment:
We can't set an ancestor id for an unlocked package, so we should skip the automatic lookup if package_type is Unlocked

Reply:
I went ahead and returned an empty string if we detect an unlocked package. I think I really should test this with a sample unlocked package, unless you have confirmed that setting an empty string as an ancestor Id doesn't throw an error. 

Alternatively, I could alter the code so that we only set a value for ancestor Id on packages that are not of type `Unlocked`.

Reply:
We've been setting ancestorId to an empty string for all 2gp package versions prior to this PR, so I'm not too worried about it.